port/espressif/esp/src/hal/time.zig

```diff
@@ -0,0 +1,81 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const time = microzig.drivers.time;
+
+// NOTE: esp32c3 specific
+const SYSTEM = microzig.chip.peripherals.SYSTEM;
+const SYSTIMER = microzig.chip.peripherals.SYSTIMER;
+
+pub fn initialize() void {
+    enable_systimer(0);
+    enable_systimer(1);
+}
+
+pub fn get_time_since_boot() time.Absolute {
+    return @enumFromInt(get_time());
+}
+
+pub fn sleep_ms(time_ms: u32) void {
+    sleep_us(time_ms * 1000);
+}
+
+pub fn sleep_us(time_us: u64) void {
+    const end_time = time.make_timeout_us(get_time_since_boot(), time_us);
+    while (!end_time.is_reached_by(get_time_since_boot())) {}
+}
+
+pub fn enable_systimer(unit_id: u1) void {
+    // This is 1 on reset, but let's set it anyway
+    SYSTEM.PERIP_CLK_EN0.modify(.{
+        .SYSTIMER_CLK_EN = 1,
+    });
+    switch (unit_id) {
+        0 => SYSTIMER.CONF.modify(.{
+            .TIMER_UNIT0_WORK_EN = 1,
+            .CLK_EN = 1,
+        }),
+        1 => SYSTIMER.CONF.modify(.{
+            .TIMER_UNIT1_WORK_EN = 1,
+            .CLK_EN = 1,
+        }),
+    }
+}
+
+pub fn get_systimer_counter(unit_id: u1) u64 {
```

Comment:
maybe comptime?

Reply:
I'd leave it for now, I don't think it would save codegen much, and making it comptime is restrictive.